lib/omise/OmiseCapabilities.php

```diff
@@ -0,0 +1,192 @@
+     *
+     * @return array
+     */
+    public function getBackends()
+    {
+        $backends = array_map(
+            function ($backend) {
+                $id = array_keys($backend)[0];
+                $backend[$id]['_id'] = $id;
+                return (object)reset($backend);
+            },
+            $this['payment_backends']
+        );
+
+        // check for filters
+        if ($filters = func_get_args()) {
+            $filter   = self::combineFilters(self::argsToVariadic($filters));
+            $backends = array_filter($backends, $filter);
+            sort($backends);
+        }
+
+        return $backends;
+    }
+
+    /**
+     * Makes a filter function to check supported currency for backend.
+     *
+     * @param  string $currency
+     *
+     * @return function
+     */
+    public function makeBackendFilterCurrency($currency)
+    {
+        return function ($backend) use ($currency) {
+            return in_array(strtoupper($currency), $backend->currencies);
+        };
+    }
+
+    /**
+     * Makes a filter function to check type of backend.
+     *
+     * @param  string $type
+     *
+     * @return function
+     */
+    public function makeBackendFilterType($type)
+    {
+        return function ($backend) use ($type) {
+            return $backend->type == $type;
+        };
+    }
+
+    /**
+     * Makes a filter function to check if backends can handle given amount.
+     *
+     * @param  int $amount
+     *
+     * @return function
+     */
+    public function makeBackendFilterChargeAmount($amount)
+    {
+        $defMin = $this['limits']['charge_amount']['min'];
+        $defMax = $this['limits']['charge_amount']['max'];
+        return function ($backend) use ($amount, $defMin, $defMax) {
+            // temporary hack for now to correct min value for instalments to 500000
+            if ($backend->type == 'installment') {
+                $min = 500000;
+            } else {
+                $min = empty($backend->amount['min']) ? $defMin : $backend->amount['min'];
+            }
+            $max = empty($backend->amount['max']) ? $defMax : $backend->amount['max'];
+            return $amount >= $min && $amount <= $max;
+        };
+    }
+
+    /**
+     * Sets up 'shortcuts' to filters so they may be used thus:
+     *    $capabilities->backendFilter['currency']('THB')
+     * As well as the original:
+     *    $capabilities->makeBackendFilterCurrency('THB')
+     */
+    protected function setupFilterShortcuts()
+    {
+        foreach (self::FILTERS as $filterSubject => $availableFilters) {
+            $filterArrayName = $filterSubject . 'Filter';
+            $this->$filterArrayName = array();
+            $tempArr = &$this->$filterArrayName;
+            foreach ($availableFilters as $type) {
+                $funcName = 'make' . ucfirst($filterSubject) . 'Filter' . $type;
+                $tempArr[$type] = function () use ($funcName) {
```

Comment:
Why spread a single operation function across so many lines?

Reply:
PHP Code Sniffer reason (same as the one I commented above).

Reply:
Drop this rule from our CodeSniffer ruleset?